### PR TITLE
[51] Do not share ResteasyProviderFactory and Registry among applicat…

### PR DESCRIPTION
**Pull request by asoldano**
…ions

Replaces https://github.com/paypal/resteasy-spring-boot/pull/52

### CODE REVIEW

File: resteasy-spring-boot-starter/src/main/java/com/paypal/springboot/resteasy/ResteasySpringBootConfig.java [hunk 1]
```diff
@@ -35,17 +35,15 @@
 
 	private static Logger logger = LoggerFactory.getLogger(ResteasySpringBootConfig.class);
 
-	private static ResteasyProviderFactory resteasyProviderFactory = new ResteasyProviderFactory();
```
Comment (fabiocarvalho777) [inline resteasy-spring-boot-starter/src/main/java/com/paypal/springboot/resteasy/ResteasySpringBootConfig.java]: This Spring bean is a singleton, so there is no need to change the scope of resteasyProviderFactory and resourceMethodRegistry. In the end of the day, the change you did in method springBeanProcessor makes no difference. Method springBeanProcessor is called only once (there will be only one bean of type BeanFactoryPostProcessor created by springBeanProcessor).

Comment (asoldano) [inline resteasy-spring-boot-starter/src/main/java/com/paypal/springboot/resteasy/ResteasySpringBootConfig.java]: @fabiocarvalho777 , have you tried the example at https://github.com/gabrielruiu/resteasy-spring-boot-incorrect-spring-context / http://stackoverflow.com/questions/41909564/resteasy-spring-incorrect-app-instance-loaded-during-tests-with-multiple-runnin ? That explain why you should use a new registry each time.

Comment (asoldano) [inline resteasy-spring-boot-starter/src/main/java/com/paypal/springboot/resteasy/ResteasySpringBootConfig.java]: Also check https://issues.jboss.org/browse/RESTEASY-1595?focusedCommentId=13355947&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13355947

Comment (fabiocarvalho777) [inline resteasy-spring-boot-starter/src/main/java/com/paypal/springboot/resteasy/ResteasySpringBootConfig.java]: Yes, I had not read the issue in stack overflow, so I take that back. Now I got what is going on. Thanks a lot for reporting it quickly and for the PR too. I am going to merge it and release a new version today.

**Pull request merged**